number_parser/parser.py

```diff
@@ -1,144 +1,136 @@
 import re
-import string
-# The initial global dictionaries that have been declared below will be changed,
-# should be easily derived automatically from CLDR data/other source for each of the languages.
-
-# We will need to have all values from 1-100 in a dictionary for some languages like hindi while others
-# english / french build 2 digit numbers using MTEN dictionary values.
-# (Difference observable in hi.json and en.json in numeral_language_data)
-
-MULTIPLIERS = {
-    "thousand": 1000,
-    "thousands": 1000,
-    "million": 1000000,
-    "millions": 1000000,
-    "billion": 1000000000,
-    "billions": 1000000000,
-    "trillion": 1000000000000,
-    "trillions": 1000000000000,
-}
-
-VALID_TOKENS_IN_NUMBERS = ["and", "-"]
-
-UNITS = {
-    word: value
-    for value, word in enumerate(
-        "one two three four five six seven eight nine".split(), 1
-    )
-}
-
-STENS = {
-    word: value
-    for value, word in enumerate(
-        "ten eleven twelve thirteen fourteen fifteen sixteen seventeen eighteen nineteen".split(), 10
-    )
-}
-
-MTENS = {
-    word: value * 10
-    for value, word in enumerate(
-        "twenty thirty forty fifty sixty seventy eighty ninety".split(), 2
-    )
-}
-
-HUNDRED = {"hundred": 100, "hundreds": 100}
-
-ALL_WORDS = {**UNITS, **STENS, **MTENS, **HUNDRED, **MULTIPLIERS}
-BASE_NUMBERS = {**UNITS, **STENS}
-
-def check_validity(current_token, previous_token):
+from importlib import import_module
+SENTENCE_SEPERATORS = [".", ","]
+
+
+class Language:
+    """Main language class to populate the requisite language-specific variables."""
+    unit_numbers = {}
+    direct_numbers = {}
+    tens = {}
+    hundreds = {}
+    big_powers_of_ten = {}
+    skip_tokens = []
+    all_numbers = {}
+    unit_and_direct_numbers = {}
+
+    def __init__(self, lang):
```

Comment:
I would check here if the language is supported to avoid:

* Using a language without tested support. Example: `lang='fr'`.
* Using non-supported values. Example `lang=222` or `lang="sss"`. (Now throws `ModuleNotFoundError`)
* Using wrong words for language (example: `lang="English"` instead of `lang="en"`. (Now throws `ModuleNotFoundError`) 

To do it, I would add:
```python
if lang not in SUPPORTED_LANGUAGES:
    raise ValueError(f'"{lang}" is not a supported language')
```

just before the `language_info = ...`

And this:
```python
SUPPORTED_LANGUAGES = ['en', 'es', 'hi', 'ru']` 
```

below the `SENTENCE_SEPARATOR` list.

(You can also leave only "en" before adding the language tests... up to you).